src/engagement/engagement-config.ts

```diff
@@ -0,0 +1,83 @@
+import * as fs from 'fs'
+import * as path from 'path'
+import * as core from '@actions/core'
+import * as yaml from 'js-yaml'
+
+/**
+ * Configuration interface for engagement scoring weights
+ */
+export interface EngagementWeights {
+  comments?: number
+  reactions?: number
+  contributors?: number
+  lastActivity?: number
+  issueAge?: number
+  linkedPullRequests?: number
+}
+
+/**
+ * Configuration interface for the .triagerc.yml file
+ */
+export interface TriageConfig {
+  engagement?: {
+    weights?: EngagementWeights
+  }
+}
+
+/**
+ * Default engagement weights that match the current hardcoded values
+ */
+export const DEFAULT_ENGAGEMENT_WEIGHTS: Required<EngagementWeights> = {
+  comments: 3,
+  reactions: 1,
+  contributors: 2,
+  lastActivity: 1,
+  issueAge: 1,
+  linkedPullRequests: 2
+}
+
+/**
+ * Load triage configuration from .triagerc.yml or .github/.triagerc.yml
+ * @param workspacePath - The workspace path to search for config files
+ * @returns Combined configuration with defaults applied
+ */
+export async function loadTriageConfig(workspacePath: string = '.'): Promise<Required<EngagementWeights>> {
+  const configPaths = [
+    path.join(workspacePath, '.triagerc.yml'),
+    path.join(workspacePath, '.github', '.triagerc.yml')
+  ]
+
+  let config: TriageConfig = {}
+
+  for (const configPath of configPaths) {
+    try {
+      if (fs.existsSync(configPath)) {
```

Comment:
Never use existsSync as this cannot be mocked nicely. Use the readFile or readdir and try/catch.